packages/phone-number-privacy/signer/src/signing/query-quota.ts

```diff
@@ -112,35 +114,56 @@ async function getQueryQuota(account: string, hashedPhoneNumber?: string) {
   ) {
     logger.debug({ account }, 'Account is not verified but meets min balance')
     // TODO consider granting these unverified users slightly less queryPerTx
-    const transactionCount = await getTransactionCountFromAccount(account)
+    const transactionCount = await getTransactionCount(account, walletAddress)
     return config.quota.unverifiedQueryMax + config.quota.queryPerTransaction * transactionCount
   }
 
   logger.debug({ account }, 'Account does not meet query quota criteria')
   return 0
 }
 
-export async function getTransactionCountFromAccount(account: string): Promise<number> {
-  return retryAsyncWithBackOff(
-    () => getContractKit().web3.eth.getTransactionCount(account),
-    RETRY_COUNT,
-    [],
-    RETRY_DELAY_IN_MS
-  )
+export async function getTransactionCount(...addresses: string[]): Promise<number> {
+  return Promise.all(
+    addresses.map((address) =>
+      retryAsyncWithBackOff(
+        () => getContractKit().web3.eth.getTransactionCount(address),
+        RETRY_COUNT,
+        [],
+        RETRY_DELAY_IN_MS
+      )
+    )
+  ).then((values) => values.reduce((a, b) => a + b))
 }
 
-export async function getDollarBalance(account: string): Promise<BigNumber> {
-  return retryAsyncWithBackOff(
-    async () => (await getContractKit().contracts.getStableToken()).balanceOf(account),
-    RETRY_COUNT,
-    [],
-    RETRY_DELAY_IN_MS
-  )
+export async function getDollarBalance(...addresses: string[]): Promise<BigNumber> {
+  return Promise.all(
+    addresses.map((address) =>
+      retryAsyncWithBackOff(
+        async () => (await getContractKit().contracts.getStableToken()).balanceOf(address),
+        RETRY_COUNT,
+        [],
+        RETRY_DELAY_IN_MS
+      )
+    )
+  ).then((values) => values.reduce((a, b) => a.plus(b)))
+}
+
+export async function getCeloBalance(...addresses: string[]): Promise<BigNumber> {
+  return Promise.all(
+    addresses.map((address) =>
+      retryAsyncWithBackOff(
+        async () => (await getContractKit().contracts.getGoldToken()).balanceOf(address),
+        RETRY_COUNT,
+        [],
+        RETRY_DELAY_IN_MS
+      )
+    )
+  ).then((values) => values.reduce((a, b) => a.plus(b)))
 }
 
-export async function getCeloBalance(account: string): Promise<BigNumber> {
+export async function getWalletAddress(account: string): Promise<string> {
   return retryAsyncWithBackOff(
-    async () => (await getContractKit().contracts.getGoldToken()).balanceOf(account),
+    async () => (await getContractKit().contracts.getAccounts()).getWalletAddress(account),
```

Comment:
So if `account` is an EOA, this will return undefined? And undefined is acceptable in all current quota methods (`getTransactionCount`, `getDollarBalance`, `getCeloBalance`) which makes it okay to do? Perhaps you can add a note in the comments since this would be hard to follow if someone didn't understand the context.